gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2333,13 +2340,52 @@ func checkOrConvertType(store Store, last BlockNode, x *Expr, t Type, autoNative
 					// default:
 				}
 			}
+			// convert x to destination type t
 			cx := Expr(Call(constType(nil, t), *x))
 			cx = Preprocess(store, last, cx).(Expr)
 			*x = cx
+		} else {
+			// if one side is declared name type and the other side is unnamed type
+			if isNamedConversion(xt, t) {
+				// covert right (xt) to the type of the left (t)
+				cx := Expr(Call(constType(nil, t), *x))
```

Comment:
Consider defining a new function for these three lines and also use it to replace the same three lines right above here to avoid potential issues in the future where someone may edit code in one spot but not the other.

Reply:
thanks, sounds good.